backend/schema/message.py

```diff
@@ -1,34 +1,96 @@
 import asyncio
 import concurrent.futures
 from datetime import datetime
-from typing import List
+from typing import Any, List
 
 from fastapi import HTTPException, status
 from pydantic import AnyHttpUrl, BaseModel, Field, root_validator
 from utils.room_utils import get_room
 
 
-class Reaction(BaseModel):
-    """Provides the nested object for reactions to message"""
+class Emoji(BaseModel):
+    """
+    Provides the nested object for reactions to message
+    """
 
-    sender_id: str
-    character: str
+    name: str
+    count: int
+    emoji: str
+    reactedUsersId: List[str] = []
 
 
 class MessageRequest(BaseModel):
     """
     Provides a base model for all threads
     """
 
-    text: str
     sender_id: str
-    reactions: List[Reaction] = []
+    emojis: List[Emoji] = []
+    richUiData: Any = {}
     files: List[AnyHttpUrl] = []
     saved_by: List[str] = []
-    edited: bool = False
+    timestamp: int = int(datetime.utcnow().timestamp())
     created_at: str = str(datetime.utcnow())
```

Comment:
There's no field to hold the text message. As a result, when we test the endpoint, we can actually send a payload successfully without a text message. 
It should not be possible, right?

Reply:
@zxenonx There's actually field for text in the richUIdata but yes I share your concern about using Any to send anything. 

We need to have a discussion, possibly with Aniebiet, regarding the current structure that's sent as the payload.

Reply:
yeah, @pauline-banye, the text field is in `richUIdata>blocks`.
So maybe for testing purposes, we could change the actual structure to this one:
```
{
  "sender_id": "619ba4671a5f54782939d385",
  "emojis": [],
  "richUiData": {,
      "blocks": [
                        {
                        "text": "HI, I'm mark.. new here",
                        }
                    ], 
},
  "files": [],
  "saved_by": [],
  "timestamp": 1640261761,
  "created_at": "2021-12-23 13:16:01.273083"
}
```
And `blocks` could be a nested schema inside richUIdata. Too messy?

Reply:
I'ld rather we get on a call for this esp with the team so as to avoid going back n forth on this. Let me know when you're free.

Reply:
@AI-fae @mykie88 @man-of-mind @o4codes

Reply:
I agree, using "any" to receive data could be a weak security point. A couple of things could go wrong.


@zxenonx , it should be possible to send a message without a text. Using WhatsApp and Slack as an example, you can send a file attachment or an emoji character without any text.

Reply:
@AI-fae very true re the sending of a file as attachment without any text. 

Yeah I do have concerns about security but based off our conversation in the group, and the reluctance to create non reusable classes to map every single thing, I believe we've decided to stick with Any. 

@zxenonx @man-of-mind @o4codes @mykie88 @kinglighthill

Reply:
I think we are all in agreement that mapping everything is not ideal. So "Any" is fine and from the assignment operator, it is by default a dictionary I guess, so we know the kind of data to expect. Besides these, I can't think of any security issue that can ensue from the current set up.
That's my take though...
@zxenonx @pauline-banye

Reply:
richUIdata is by default a dict, so, `dict` would be fine for the type instead of `Any`.
@pauline-banye